src/GovUk.Education.ExploreEducationStatistics.Data.Services/AbstractSubjectMetaService.cs

```diff
@@ -124,9 +79,25 @@ protected static List<IndicatorMetaViewModel> BuildIndicatorViewModels(IEnumerab
             });
         }
 
-        private string GetTotalValue(IEnumerable<FilterItem> filterItems)
+        protected Task<Either<ActionResult, ReleaseSubject>> CheckReleaseSubjectExists(Guid releaseId, Guid subjectId)
         {
-            return _filterItemRepository.GetTotal(filterItems)?.Id.ToString() ?? string.Empty;
+            return _persistenceHelper.CheckEntityExists<ReleaseSubject>(
+                query => query
+                    .Include(rs => rs.Subject)
+                    .Where(rs => rs.ReleaseId == releaseId
+                                 && rs.SubjectId == subjectId)
+            );
+        }
+
+        protected async Task<Either<ActionResult, ReleaseSubject>> CheckCanViewSubjectData(
+            ReleaseSubject releaseSubject)
+        {
+            if (await _userService.MatchesPolicy(releaseSubject.Subject, DataSecurityPolicies.CanViewSubjectData))
```

Comment:
Would it be a good idea to put this on a `UserServiceExtensions` method like we do for the content and admin security policies?

Reply:
Ah yeah, good idea, this makes things one step closer to getting rid of this abstract class. Thanks.